### DISCUSSION TITLE
Can Shiny Jobs fetch data when an app is closed?

### DISCUSSION THREAD

**Discussion by simspace-dev** [shinyorg/shiny#1270] [Q&A] [answered]
I need to update a Xamarin app (iOS and Android) to fetch and cache data while the app is closed (not backgrounded). The app uses several backend service APIs to load various types of data. For starters, I was researching iOS Background Fetching and NSURLSession and it was during that research that I discovered Shiny.

Can Shiny be used fetch data via backend service APIs when the app is closed (not backgrounded)? 

Some posts I have read have led me to believe it can.  Other posts that mention `BGTaskRequest` have led me to believe it cannot. It's my understanding that `BGTaskRequest` and the related classes only work when the app is running in either the foreground or background, but not when the app is closed. 

Can you help clear up my confusion?

Thank you!

**Comment by aritchie** [comment]
BGTaskRequest will run (at some point) if the app is suspended.  If the app is killed (swiped away), no - it won't run.

**Reply by simspace-dev** [reply]
That's what I thought, thanks for clearing that up. 
Is that the case with Shiny Jobs? Or is Shiny able to fetch data when the app is killed?

**Reply by aritchie** [reply, marked as answer]
Shiny can't break the rules of the platform

**Reply by simspace-dr** [reply]
Sounds good, thank you!
